tests/test_scalars.py

```diff
@@ -517,6 +517,7 @@ def test_timedelta_properties_methods() -> None:
     check(assert_type(td.value, int), int)
     check(assert_type(td.resolution_string, str), str)
     check(assert_type(td.components, Components), Components)
+    check(assert_type(td.unit, str), str)
```

Comment:
Since `TimeUnit` is in `pandas`, you could import that and use it here in the `assert_type` test.